src/aks-preview/azext_aks_preview/agentpool_decorator.py

```diff
@@ -867,6 +885,17 @@ def set_up_skip_gpu_driver_install(self, agentpool: AgentPool) -> AgentPool:
             agentpool.gpu_profile.install_gpu_driver = False
         return agentpool
 
+    def set_up_driver_type(self, agentpool: AgentPool) -> AgentPool:
+        """Set up driver type property for the AgentPool object."""
+        self._ensure_agentpool(agentpool)
+
+        driver_type = self.context.get_driver_type()
+        if driver_type is not None:
+            if agentpool.gpu_profile is None:
+                agentpool.gpu_profile = self.models.AgentPoolGPUProfile()  # pylint: disable=no-member
+            agentpool.gpu_profile.driver_type = driver_type
```

Comment:
what if driver_type is explicitly provided meanwhile with  --skip-gpu-driver-install, will this fail in RP FE?

Reply:
RP FE will reject with IllegalGPUDriverTypeWhenInstallFalse